### PR TITLE
Contribution from Netherlands eScience Center

### PR DESCRIPTION
**Pull request by rogerkuou**
This PR includes the course refactoring from the Netherlands eScience Center. The contributions are made by @fnattino @SarahAlidoost @cpranav93 @raar1 @rogerkuou.

Summary of the changes:

1. The Episodes are restructured. 
    - There are now 11 Episodes available, 1 still [WIP](https://github.com/esciencecenter-digital-skills/geospatial-python/issues/9). 
    - Episodes 1 to 4 remain mostly the same. 
    - Episode 5 is a new episode, focusing on satellite data retrival.
    - Episode 6 to 10  are inspired by the current material, focusing on the same Python tools and trying to cover the same learning goals.
    - Episode 11 is another new episode,  focusing on parallel computation for raster data. 
2.  To make the example case a bit more eye catchy, we changed the AoI to Northern Amsterdam, because the ground coverage has more varieties in this location. We also replaced the NEON Lidar datasets with the following open data:
    - Raster: Sentinel-2 optical image
    - Vector Three [PDOK ](https://www.pdok.nl/) cadaster data:
        -  * [Polygon: Crop fields data](https://service.pdok.nl/rvo/brpgewaspercelen/atom/v1_0/downloads/brpgewaspercelen_definitief_2020.gpkg)
       * [Points: Ground wate](https://service.pdok.nl/bzk/brogmwvolledigeset/atom/v2_1/downloads/brogmwvolledigeset.zip)
       * [Polyline: Waterways](https://geo.rijkswaterstaat.nl/services/ogc/gdr/vaarweginformatie/ows?service=WFS&version=2.0.0&request=GetFeature&typeName=status_vaarweg&outputFormat=SHAPE-ZIP)
3. Miscellaneous improvements:
    - Data has been backed up on Figshare: https://doi.org/10.6084/m9.figshare.20146919.v1
    - Restructured the figures. Now all the figures are in the `fig` directory, with the naming convention is 'episode_number-image_number_in_episode-name_of_image', i.e. E01-01-imagename.jpg
    - Added an initial version of instructor notes

### CODE REVIEW

**Comment by rbavery** [comment]
Fantastic. Thank you @fnattino @SarahAlidoost @cpranav93 and @rogerkuou for pushing this lesson forward and overhauling each of these lessons to make it a modern geospatial python lesson.

I'll be reviewing this throughout the month. can you remind me when you intend to teach this version?

**Comment by rbavery** [comment]
@rogerkuou can we remove all the files in the `_episodes/_extras` folder? are these older versions that are now numbered and exist in `_episodes`? Or do they have another purpose? If they should be kept, I'm wondering if I should review these at all or if I should focus on the top level `_episodes` folder.

**Comment by rogerkuou** [comment]
> @rogerkuou can we remove all the files in the `_episodes/_extras` folder? are these older versions that are now numbered and exist in `_episodes`? Or do they have another purpose? If they should be kept, I'm wondering if I should review these at all or if I should focus on the top level `_episodes` folder.

Hi @rbavery, good point. I have removed the folder `_episodes/_extras` since they are only initial templates. On a related token, I would tend to keep `_episodes/_legacy` for now since it contains the previous version of some episodes. Just in case we need to have something back.

**Comment by rogerkuou** [comment]
> Fantastic. Thank you @fnattino @SarahAlidoost @cpranav93 and @rogerkuou for pushing this lesson forward and overhauling each of these lessons to make it a modern geospatial python lesson.
> 
> I'll be reviewing this throughout the month. can you remind me when you intend to teach this version?

And my apology for missing @raar1 in the initial PR description. Sorry Robin! I  know you won't mind but it's not that Ryan excluded you on purpose!

**Comment by rbavery** [comment]
@rogerkuou @fnattino I'm almost done reviewing but wanted to point out a blocker to merging.

[Polygon: Crop fields data](https://service.pdok.nl/rvo/brpgewaspercelen/atom/v1_0/downloads/brpgewaspercelen_definitief_2020.gpkg) is too large to download. since it is a half gigabyte and is located on a PDOK server with a slow bandwidth connection or the fact that it is located in the Netherlands, it would take an hour for me to download this file on the USA west coast. Details here, maybe we can discuss a solution in this issue: https://github.com/esciencecenter-digital-skills/geospatial-python/issues/61

**Review by rbavery — APPROVED** [review]
Awesome job y’all. I’ve reviewed the lesson and it looks fantastic. I’ve made a few commits to fix missing imports, update the data file for a manageable download, and rephrase some language, but I saw nothing substantial that needed to change. This is an amazing revamp of the lesson and I’m very excited to teach it later this month.
I left review comments but none of these should hold this back from a merge I think. I can go through and make the language/grammar updates I noticed at a later date. I’d like to merge this by the end of this coming week if everyone approves. That way I can give folks that will be attending Maptime Davis time to review the material and setup instructions (since it will be a relatively short workshop).

One final thing, I wasn't able to resolve the merge conflicts on my end. I'm unable to git fetch the upstream on the escience repo because of a [possible large file issue](https://superuser.com/questions/1205429/git-hangs-at-fetch-but-runs-fine-with-pull) which exists somewhere in the git record.  I also can't do this on the upstream repo end, when I try to git push my merge edits after `gh pr checkout` I hit a different error. @rogerkuou or others, could one of you take a crack at resolving the merge conflicts? Both conflicts can be resolved by accepting the changes in this current pr and ignoring what is on main `gh-pages`

File: _episodes/07-vector-data-in-python.md
Comment (fnattino) [reply _episodes/07-vector-data-in-python.md]: Good point @rbavery ! I have modified the implementation, using a lambda function to keep it concise (it might also be a good way to introduce these) - what do you think?

**Comment by fnattino** [comment]
Hi @rbavery , thank you very much for reviewing this big PR and for raising and addressing [the issue on the size of the vector dataset](https://github.com/esciencecenter-digital-skills/geospatial-python/issues/61). I should have implemented all your suggestions - please merge when/if you deem this is ready!

**Pull request merged**
